### PR TITLE
feat: add workshop for building hash commitments

### PR DESCRIPTION
**Pull request by 0xpanoramix**
Signed-off-by: Luca Georges Francois <luca@quartz.technology>

This PR adds the subject for building a Hash Commitment Protocol using Python.

### CODE REVIEW

File: cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+## Step 0: Initialization
+
+Please make sure to follow the instructions given in the [SETUP.md](./SETUP.md) file before you start the workshop.
+
+## Step 1: Prepare Commitment Phase
+
+### 📑 Description
+
+First things first, both the two players must make their choices. The object selection is already implemented
```
Comment (TomChv) [inline cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md:68]: `First things first` 👀

File: cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+### 📑 Description
+
+In the first step, you created a random number used in the second step to create a commitment for each of the two players choices.
+In this last step, you will verify those commitments based on what object each player claims he has chosen.
+
+Put in simple words, each player will create a commitment for the other player's choice using his secret random number, that he will now reveal (take a look at the `Player.tell_choice` and `Player.reveal` methods).
+
+Example:
+1. Player 2 uses Player's 1 initial commitment, claim and secret to create a verification commitment.
```
Comment (TomChv) [inline cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md:188]: I do not understand "claim and secret"

File: cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+Player 1 is lying 😈
+```
+
+### 📚 Documentation
+
+- [Use the hashlib library to compute the hash of a string](https://datagy.io/python-sha256/)
+
+## Conclusion
+
+That's it ! You have created a version of the rock paper scissors game which is secured by the use of hash commitments.
```
Comment (TomChv) [inline cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md:252]: No "To go further" ?

File: cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/rps_base.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+class Object(Enum):
+    """Represents one of the possible object that can be chosen by a player."""
+
+    ROCK = "rock"
+    PAPER = "paper"
+    SCISSORS = "scissors"
+
+
+class Player:
+    """Represents one player and stores it's choice."""
```
Comment (TomChv) [inline cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/rps_base.py:16]: "he's choice"? It's a person no?

**Review by TomChv — COMMENTED** [review]
Nice workshop, I left few comments :rocket:

File: cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md (hunk 2 above)
Comment (0xpanoramix) [reply cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md:188]: The initial commitment, the claim and the secret belonging to the Player 1.

File: cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md (hunk 3 above)
Comment (0xpanoramix) [reply cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md:252]: Nope not here ! I've already left a comment teasing Part 2 with Pedersen Commitments.

File: cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+
+The first way to achieve this is by preventing the participants to cheat in the first place.
+How ? They will have to reveal their choice to each other at the same time. Unfortunately, this is
+not achievable due to network synchronization as requests on real-world networks are not
+instantaneous.
+
+Instead, we will use cryptographic commitments to force the participants to stick with their initial
+choices, otherwise their misbehavior will be exposed to everyone.
+
+In this workshop, we will implement hash commitments using Python for our rock paper scissors game.
```
[the remark this answers is not in the document]
Comment (0xpanoramix) [reply cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md]: I think it's a good way to introduce the achievements of this workshop without saying too much about the protocol.

[the remark this answers is not in the document]
Comment (0xpanoramix) [reply cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md]: Fixed in b893cd7902a591561be4f956087e40fa61a7c01c

File: cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/rps_base.py (hunk 4 above)
Comment (0xpanoramix) [reply cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/rps_base.py]: Fixed in b893cd7902a591561be4f956087e40fa61a7c01c

File: cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md (hunk 2 above)
Comment (0xpanoramix) [reply cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md]: Fixed in b893cd7902a591561be4f956087e40fa61a7c01c

File: cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+
+## Step 0: Initialization
+
+Please make sure to follow the instructions given in the [SETUP.md](./SETUP.md) file before you start the workshop.
+
+## Step 1: Prepare Commitment Phase
+
+### 📑 Description
+
+First things first, both the two players must make their choices. The object selection is already implemented
```
Comment (TomChv) [inline cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md:67]: What "first things first" mean?

**Review by TomChv — APPROVED** [review]
Left one last comment but you can merge if it's not relevant.

File: cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md (hunk 6 above)
Comment (0xpanoramix) [reply cryptography/0x02 - Cryptographic Commitments/Part 1 - Hash Commitments/README.md:67]: "Commençons par le commencement"

**Pull request merged**